docs/source/reference/strings.rst

```diff
@@ -0,0 +1,137 @@
+
+Multiline string literals
+--------------------------
+
+In some cases you will have to display a large string literal that spans multiple lines. For this you
+can use *multiline string literals*, they allow you to span a string across multiple vertical
+lines, preserving the line returns and the indentation. Additionally they allow you to indent your
+multiline string with the surrounding code, without breaking the intended format of the string.
+
+To use multiline strings, start with a triple quote ``"""`` followed by a line return, then
+enter your text and close it with another triple quote ``"""`` with whitespace on its left.
+The indentation of the closing triple quote will determine how much whitespace should be cropped
+from each line of the text.
+
+.. note::
+
+   Multiline strings use triple quotes to enable the automatic cropping of leading whitespace
+   when the multiline block is indented.
+
+
+.. code-block:: idris
+
+    welcome : String
+    welcome = """
+        Welcome to Idris 2
+
+        We hope you enjoy your stay
+          This line will remain indented with 2 spaces
+        This line has no intendation
+        """
+
+printing the variable `welcome` will result in the following text:
+
+::
+
+    Welcome to Idris 2
+
+    We hope you enjoy your stay
+      This line will remain indented with 2 spaces
+    This line has no intendation
+
+As you can see, each line has been stripped of its leading 4 space, that is because the closing
+delimiter was indented with 4 spaces.
+
+In order to use multiline string literals, remember the following:
+
+- The starting delimited must be followed by a line return
+- The ending delimiter's intendation level must not exceed the indentation of any line
+
+Raw string literals
+-------------------
+
+It is not uncommon to write string literals that require some amount of escaping. For plain string
+literals the characters ``\\`` and ``"`` must be escaped, for multiline strings the characters
+``"""`` must be escaped. Raw string literals allow you to dynamically change the required
+escaped
+sequence in order to avoid having to escape those very common sets of characters. For this, use
+``#"`` as starting delimiter and ``"#`` as closing delimiter. The number of ``#`` symbols can be
+increased in order to accomodate for edge cases where ``"#`` would be a valid symbol.
+In the following example we are able to match on ``\{`` by using half as many ``\\`` characters
+as if we didn't use raw string literals:
+
+.. code-block:: idris
+
+    myRegex : Regex
+    myRegex = parseRegex #"\\{"#
+
+If you need to escape characters you still can by using a ``\\`` followed by the same number of
+``#`` that you used for your string delimiters. In the following example we are using two
+``#`` characters as our escape sequence and want to print a line return:
+
+.. code-block::
+
+    markdownExample : String
+    markdownExample = ##"markdown titles look like this: \##n"# Title \##n body""##
```

Comment:
I'm confused: why do you have `\##n` instead of `\n`?
According to the next example using a multiline string you *do* want the newline?

Reply:
Because the escape sequence is `\##` in that string, if you were to write `\n` the _text_ `\n` would appear instead of a new line.

Reply:
I see. I always understood `\n` as a single character, not as `n` being escaped to mean newline. :thinking:

Reply:
the character `n` preceeded with the escape sequence renders a single character, just like `\t`, `\r`, `\0`. Raw strings allow the user to type those character without having to escape the `\` character. But you can recover their meaning by using the `\#` escape sequence inside a raw string (or however many `#` you're using for the raw string)